Sources/SwiftFormatRules/TypeNamesShouldBeCapitalized.swift

```diff
@@ -0,0 +1,52 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftFormatCore
+import SwiftSyntax
+
+/// `struct`, `class`, `enum` and `protocol` declarations should have a capitalized name.
+///
+/// Lint:  Types with un-capitalized names will yield a lint error.
+public final class TypeNamesShouldBeCapitalized : SyntaxLintRule {
```

Comment:
Can you also add `ActorDeclSyntax`, `AssociatedTypeDeclSyntax`, and `TypeAliasDeclSyntax`? I think that covers everything we'd want to apply this to.

Reply:
Sure! Will do! I went conservative initially to not over do it :)